lib/Parse/ParseDecl.cpp

```diff
@@ -5778,16 +5778,19 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
         Status.setIsParseError();
         return Status;
       }
-      if (CommaLoc.isValid()) {
-        diagnose(Tok, diag::expected_identifier_after_case_comma);
-        Status.setIsParseError();
-        return Status;
-      }
+
       if (NameIsKeyword) {
         diagnose(TokLoc, diag::keyword_cant_be_identifier, TokText);
         diagnose(TokLoc, diag::backticks_to_escape)
           .fixItReplace(TokLoc, "`" + TokText.str() + "`");
+        if (Tok.isNot(tok::kw_case))
```

Comment:
If the keyword is *not* on the same line, that may be a start of another decl. e.g.
```swift
enum E {
  case foo
  case 
 
  func bar() { }
}
```
So, let's keep it if `Tok.isAtStartOfLine()`.